shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -2594,169 +2611,211 @@ impl KafkaSinkCluster {
                         .unwrap()
                         .rack
                         .as_ref();
-                    let shotover_node = self
-                        .shotover_nodes
-                        .iter()
-                        .find(|shotover_node| {
-                            coordinator_rack
-                                .map(|rack| rack == &shotover_node.rack)
-                                .unwrap_or(true)
-                        })
-                        .unwrap();
-                    coordinator.host = shotover_node.address.host.clone();
-                    coordinator.port = shotover_node.address.port;
-                    coordinator.node_id = shotover_node.broker_id;
+
+                    let hash = hash_str_bytes(coordinator.key.clone());
+                    let (shotover_nodes_in_rack, shotover_nodes_out_of_rack) =
```

Comment:
lets move this out of the loop to avoid recreating these vecs every loop.

Reply:
Actually... its very rare to actually handle multiple coordinators in a single message.
Maybe we are better off combining partition_shotover_nodes_by_rack and select_shotover_node_by_hash into as single call.

Reply:
Oh oops, I missed that partition_shotover_nodes_by_rack and select_shotover_node_by_hash are used outside of this function. Never mind then.